src/components/Login/Login.tsx

```diff
@@ -98,31 +100,28 @@ export default class LoginModal extends Component<Partial<LoginModalProps>, Logi
     });
   };
 
-  public handleUsernameChange = event => {
+  public handleUsernameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
     this.setCredentials('username', event);
   };
 
-  public handlePasswordChange = event => {
+  public handlePasswordChange = (event: React.ChangeEvent<HTMLInputElement>) => {
     this.setCredentials('password', event);
   };
 
-  public handleValidateCredentials = event => {
+  public handleValidateCredentials = (event: React.FormEvent<HTMLFormElement>) => {
     const { form } = this.state;
     // prevents default submit behavior
     event.preventDefault();
 
     this.setState(
       {
-        form: Object.keys(form).reduce(
-          (acc, key) => ({
-            ...acc,
-            ...{ [key]: { ...form[key], pristine: false } },
-          }),
-          { username: {}, password: {} }
-        ),
+        form: {
+          username: { ...form.username, pristine: false },
```

Comment:
I opted for a simple assignment since the keys weren't dynamic and the reduce construct was hard to understand for TypeScript.